_tools/fetch-team-data.js

```diff
@@ -71,25 +104,50 @@ const team = {
 
         // this user only has login, need to fetch the full profile for useful data
         for (const user of result) {
-            const { data: profile } = await octokit.users.getByUsername({ username: user.login });
-
-            team[teamIds[teamId]].push({
-                username: user.login,
-                name: profile.name,
-                website: profile.blog,
-                avatar_url: profile.avatar_url,
-                bio: profile.bio,
-                twitter_username: profile.twitter_username,
-                github_username: user.login,
-                location: profile.location
-            });
+            const profile = await fetchUserProfile(user.login);
+
+            team[teamIds[teamId]].push(profile);
         }
 
     }
 
     // filter out TSC members and reviewers from committers list
     team.committers = team.committers.filter(user => !team.tsc.find(tscmember => tscmember.username === user.username) &&
-            !team.reviewers.find(tscmember => tscmember.username === user.username));
+        !team.reviewers.find(tscmember => tscmember.username === user.username));
+
+    return team;
+}
+
+async function fetchBlogAuthors() {
+
+    const authors = {};
+    const files = (await fs.readdir(blogPostsDirectory))
+        .filter(filename => filename.endsWith(".md"))
+        .map(filename => path.join(blogPostsDirectory, filename));
+
+    for (const filename of files) {
+        const contents = await fs.readFile(filename, "utf8");
+        const { data: frontmatter } = matter(contents);
+
+        for (const username of frontmatter.authors) {
+            authors[username] = await fetchUserProfile(username);
```

Comment:
Since the frontmatter is just plain strings, we should expect some usernames will have typos or just plain not exist in GitHub. What’s the desired behavior in that case? Do we blow up so we can fix it, or skip it?

Reply:
Blow it up. We should never check in authors that don’t exist.

Reply:
:bomb: it is!